### PR TITLE
Swagger docs

### PR DESCRIPTION
**Pull request by CharlesSquirel**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
Based on the provided task requirements and the code changes, here is my review:

1. **All existing routes should be documented**: The code changes seem to cover this requirement. The Swagger configuration now includes documentation for the following endpoints: `/cards`, `/cards/author/{author}`, `/cards/tags/{tag}`, and `/cards/{id}`. Each endpoint has a summary, parameters (if applicable), security, and responses defined.

2. **Header authorization should be included**: This requirement is also met. Each endpoint in the Swagger configuration includes a `security` field with `BearerAuth: []`, which indicates that the endpoint requires Bearer Token Authorization.

However, there are some issues in the code:

- There is a missing colon after the '500' status code in the responses for the `GET /cards` endpoint.
- The indentation is inconsistent in some places, which could lead to parsing errors. For example, under the `GET /cards` endpoint, the `items` field should be indented one level more than its parent field `type: array`.
- The `description` field under the '404' status code for the `DELETE /cards/{id}` endpoint seems to be incorrect. It says "It is not allowed to delete the card after 5 minutes of its creation." which seems more like a '403 Forbidden' response.

Here are some corrections:

```diff
@@ -4,3 +4,175 @@ info:
   version: 1.0.0
 apis:
   - ./src/routes/routes
+servers:
+  -url: http://api.example.com/v1
+paths:
+  /cards:
+    get:
+      summary: Returns array of cards
+      security:
+        - BearerAuth: []
+    responses:
+      '200':
+      description: Successful operation
+      content:
+        application/json:
+          schema:
+            type: array
+              items: 
+                type: object
+                properties:
+                  front:
+                    type: string
+                    description: Text in first side of the card
+                  back:
+                    type: string
+                    description: Text in second side of the card
+                  tags:
+                    type: array
+                    items:
+                      type: string
+                    description: Tags related to the card
+                  author:
+                    type: string
+                    description: Author of the card
+        "404":
+        description: Cards not found
+        content:
+          application/json:
+            example:
+              error: "Cards not found"
+              details: "Cards not found in the database. Check database connection and make sure it is not empty"
+        "500":
+        description: Problem with the server
...
...
...
  /cards/{id}:
...
...
...
    delete:
+      summary: Delete the card
+      parameters:
+        - name: id
+          in: path
+          required: true
+          schema:
+            type: string
+          description: Card's ID
+      security:
+        - BearerAuth: []
+      responses:
+        '204':
+          description: Deleting succesfull
+        '403':
+          description: Card not found or It is not allowed to delete the card after 5 minutes of its creation.
+        '400':
+          description: Invalid card ID
```

Overall, the code changes seem to cover the task requirements, but some minor corrections are needed to ensure the Swagger configuration is correct and consistent.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task Goal
Cover existing endpoints with Swagger.

### Requirements
1. All existing routes should be documented.
2. Header authorization should be included.

### Code Changes Review

#### File: src/controllers/cardcontroller.ts
The change in this file is not related to the task requirements. It's a change in the HTTP status code from 404 to 403 when trying to delete a card after 5 minutes of its creation. This change doesn't affect Swagger documentation.

#### File: swagger-config.yaml
The changes in this file are related to the task requirements. The Swagger documentation for the routes is being added.

1. The documentation for the GET and DELETE methods for the /cards and /cards/{id} endpoints has been added. This covers the first requirement.
2. The 'BearerAuth' security scheme has been added to the GET and DELETE methods, which covers the second requirement.

However, it seems like not all routes are documented yet. The task requires all existing routes to be documented, but from the provided changes, it seems like only two routes are covered. If there are more routes in the application, they need to be documented as well.

Also, there's a typo in the servers section, `-url:` should be `- url:`.

### Suggestions
1. Document all existing routes if there are more than the ones provided in the changes.
2. Fix the typo in the servers section.

Example:
```diff
- -url: http://api.example.com/v1
+ - url: http://api.example.com/v1
```

Overall, the changes partially cover the task requirements. The Swagger documentation has been added and the header authorization is included, but potentially not all routes are documented yet.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task Goal
Cover existing endpoints with Swagger.

### Requirements
- All existing routes should be documented.
- Header authorization should be included.

### Code Review

#### src/controllers/cardcontroller.ts
The change in this file is not related to the task requirements. It changes the HTTP status code from 404 to 403 when trying to delete a card after 5 minutes of its creation. This change does not affect Swagger documentation.

#### swagger-config.yaml
The changes in this file are related to the task requirements. The Swagger documentation for the `/cards` and `/cards/{id}` endpoints have been added. 

The `get` method for `/cards` endpoint and `delete` method for `/cards/{id}` endpoint have been documented. The responses for each method, including the possible HTTP status codes, are also documented. 

The `security` field has been added to both methods, which indicates that a Bearer token is required for authorization. This covers the requirement for including header authorization.

However, it seems like not all existing routes are documented yet. The task requires all existing routes to be documented, but the provided changes only cover two methods of two endpoints. If there are more methods and endpoints in the API, they should also be included in the Swagger documentation.

### Suggestions
- Document all existing routes in the Swagger configuration file.
- Make sure to include the `security` field for all routes that require authorization.
- For each route, document all possible responses, including error responses.
- For each response, provide a description and an example if possible.

File: src/controllers/cardcontroller.ts [hunk 1]
```diff
@@ -122,7 +122,7 @@ export const deleteCard: RequestHandler = async (req, res, next) => {
     const currentTime = new Date().getTime();
```
Comment (Sumick) [inline src/controllers/cardcontroller.ts:122]: **nitpick:** should be `cardController` (written in camelCase)

**Review by Sumick — APPROVED** [review]
Looks good!

**Pull request merged**